From d9fc8710b5013711c1e1d32b8e5936fa88f85ff6 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Wed, 9 Jan 2019 18:58:45 +0100
Subject: [PATCH 4/9] Quick and dirty int13 fix.

Do not submit this patch.

For some reason cylinders becomes 0 (device size is 0? cgi-bin script
wrong?) this is a quick fix to make Windows boot.
---
 src/arch/x86/interface/pcbios/int13.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/arch/x86/interface/pcbios/int13.c b/src/arch/x86/interface/pcbios/int13.c
index ca789a0d..c1042b3f 100644
--- a/src/arch/x86/interface/pcbios/int13.c
+++ b/src/arch/x86/interface/pcbios/int13.c
@@ -398,7 +398,7 @@ static int int13_guess_geometry ( struct san_device *sandev, void *scratch ) {
 		blocks_per_cyl = ( int13->heads * int13->sectors_per_track );
 		assert ( blocks_per_cyl != 0 );
 		int13->cylinders = ( blocks / blocks_per_cyl );
-		if ( int13->cylinders > 1024 )
+		if ( int13->cylinders > 1024 || int13->cylinders == 0)
 			int13->cylinders = 1024;
 	}
 
-- 
2.17.0

